components/Splash.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { StyleSheet, Image, View } from 'react-native';
+import { NavigationActions } from 'react-navigation';
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        flexDirection: 'row',
+        backgroundColor: '#000',
+        justifyContent: 'center',
+    },
+    logoStyle: {
+        marginTop: '20%',
+        width: 150,
+        height:150,
+    },
+});
+
+let sleep = (ms) => {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+class SplashScreen extends React.Component {
+    componentDidMount() {
+        const dispatchAction = NavigationActions.reset({
+            index: 0,
+            actions: [NavigationActions.navigate({ routeName: 'App' })],
+        });
+
+        sleep(2000).then(() => {
```

Comment:
Yes. I needed a trigger event to remove the splash screen. Ideally, once we get content or some web calls for initialization this can be replaced with a watcher function/promise.